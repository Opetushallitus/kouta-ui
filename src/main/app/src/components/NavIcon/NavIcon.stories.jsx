import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import NavIcon from './index';

storiesOf('NavIcon', module)
  .add('Basic', () => <NavIcon icon="school">Koulutus</NavIcon>)
  .add('With state', () => (
    <>
      <NavIcon icon="school" done>
        Koulutus
      </NavIcon>
      <NavIcon icon="settings" active>
        Toteutus
      </NavIcon>
    </>
  ));
