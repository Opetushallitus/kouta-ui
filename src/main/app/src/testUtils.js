import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
