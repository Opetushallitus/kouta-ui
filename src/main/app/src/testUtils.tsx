import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
