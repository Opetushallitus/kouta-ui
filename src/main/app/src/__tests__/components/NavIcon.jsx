import React from 'react';

import NavIcon from '../../components/NavIcon';
import { mountWithTheme } from '../../testUtils';

test('renders correctly', () => {
  expect(
    mountWithTheme(<NavIcon icon="settings">Settings</NavIcon>),
  ).toMatchSnapshot();
});

test('renders correctly with state', () => {
  expect(
    mountWithTheme(
      <NavIcon icon="settings" active>
        Settings active
      </NavIcon>,
    ),
  ).toMatchSnapshot();

  expect(
    mountWithTheme(
      <NavIcon icon="settings" done>
        Settings done
      </NavIcon>,
    ),
  ).toMatchSnapshot();
});
