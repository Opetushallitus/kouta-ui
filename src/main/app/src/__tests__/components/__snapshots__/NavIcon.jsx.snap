// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  font-size: 1.5rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
}

.c4 {
  margin-top: 8px;
  text-align: center;
  font-family: 'Open Sans',sans-serif;
  color: #666666;
  font-size: 0.8rem;
}

.c2 {
  font-size: 2rem;
  color: #666666;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.div
      active={false}
      done={false}
    >
      <div
        className="c1"
      >
        <Styled(Icon)
          active={false}
          type="settings"
        >
          <styled.i
            active={false}
            className="material-icons c2"
          >
            <i
              className="material-icons c2 c3"
            >
              settings
            </i>
          </styled.i>
        </Styled(Icon)>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="c4"
      >
        Settings
      </div>
    </styled.div>
  </div>
</styled.div>
`;

exports[`renders correctly with state 1`] = `
.c3 {
  font-size: 1.5rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
  background-color: #2da0c7;
  border-color: white;
}

.c4 {
  margin-top: 8px;
  text-align: center;
  font-family: 'Open Sans',sans-serif;
  color: #666666;
  font-size: 0.8rem;
}

.c2 {
  font-size: 2rem;
  color: #666666;
  color: white;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.div
      active={true}
      done={false}
    >
      <div
        className="c1"
      >
        <Styled(Icon)
          active={true}
          type="settings"
        >
          <styled.i
            active={true}
            className="material-icons c2"
          >
            <i
              className="material-icons c2 c3"
            >
              settings
            </i>
          </styled.i>
        </Styled(Icon)>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="c4"
      >
        Settings active
      </div>
    </styled.div>
  </div>
</styled.div>
`;

exports[`renders correctly with state 2`] = `
.c3 {
  font-size: 1.5rem;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border-radius: 50%;
  width: 3.5rem;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid;
  background-color: white;
  border-color: #cecece;
  border-color: #2da0c7;
}

.c4 {
  margin-top: 8px;
  text-align: center;
  font-family: 'Open Sans',sans-serif;
  color: #666666;
  font-size: 0.8rem;
}

.c2 {
  font-size: 2rem;
  color: #666666;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.div
      active={false}
      done={true}
    >
      <div
        className="c1"
      >
        <Styled(Icon)
          active={false}
          type="settings"
        >
          <styled.i
            active={false}
            className="material-icons c2"
          >
            <i
              className="material-icons c2 c3"
            >
              settings
            </i>
          </styled.i>
        </Styled(Icon)>
      </div>
    </styled.div>
    <styled.div>
      <div
        className="c4"
      >
        Settings done
      </div>
    </styled.div>
  </div>
</styled.div>
`;
